cmd/get-measurements/main.go

```diff
@@ -0,0 +1,164 @@
+package main
+
+//
+// Make a HTTP GET request over a TEE-attested connection (to a server with aTLS support),
+// and print the verified measurements and the response payload.
+//
+// Currently only works for Azure TDX but should be easy to expand.
+//
+// Usage:
+//
+//   go run cmd/get-measurements/main.go --addr=https://instance_ip:port
+//
+// Can also save the verified measurements and the response body to files:
+//
+//   go run cmd/get-measurements/main.go --addr=https://instance_ip:port --out-measurements=measurements.json --out-response=response.txt
+//
+
+import (
+	"encoding/asn1"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/flashbots/cvm-reverse-proxy/common"
+	"github.com/flashbots/cvm-reverse-proxy/internal/atls"
+	"github.com/flashbots/cvm-reverse-proxy/internal/attestation/variant"
+	"github.com/flashbots/cvm-reverse-proxy/proxy"
+	"github.com/urfave/cli/v2" // imports as package "cli"
+)
+
+var flags []cli.Flag = []cli.Flag{
+	&cli.StringFlag{
+		Name:  "addr",
+		Value: "https://localhost:7936",
+		Usage: "TEE server address",
+	},
+	&cli.StringFlag{
+		Name:  "out-measurements",
+		Value: "",
+		Usage: "Output file for the measurements",
+	},
+	&cli.StringFlag{
+		Name:  "out-response",
+		Value: "",
+		Usage: "Output file for the response payload",
+	},
+	&cli.BoolFlag{
+		Name:  "log-debug",
+		Value: false,
+		Usage: "log debug messages",
+	},
+}
+
+func main() {
+	app := &cli.App{
+		Name:   "get-measurements",
+		Usage:  "Get verified measurements",
+		Flags:  flags,
+		Action: runClient,
+	}
+
+	if err := app.Run(os.Args); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func runClient(cCtx *cli.Context) (err error) {
+	logDebug := cCtx.Bool("log-debug")
+	addr := cCtx.String("addr")
+	outMeasurements := cCtx.String("out-measurements")
+	outResponse := cCtx.String("out-response")
+
+	// Setup logging
+	log := common.SetupLogger(&common.LoggingOpts{
+		Debug:   logDebug,
+		JSON:    false,
+		Service: "get-measurements",
+		Version: common.Version,
+	})
+
+	if !strings.HasPrefix(addr, "https://") {
+		return errors.New("address needs to start with https://")
+	}
+
+	log.Info("Getting verified measurements from " + addr + " ...")
+	// Prepare aTLS stuff
+	serverAttestationType := proxy.AttestationAzureTDX
+	issuer, err := proxy.CreateAttestationIssuer(log, serverAttestationType)
+	if err != nil {
+		log.Error("could not create attestation issuer", "err", err)
+		return err
+	}
+
+	validators, err := proxy.CreateAttestationValidators(log, serverAttestationType, "measurements-empty.json")
```

Comment:
why are we creating validators for an empty measurement? shouldn't the validator be created for the fetched measurement to verify/attest?

Reply:
good catch! removed

Reply:
(no validator should be there, we don't want the atls library to validate the measurements, just to get them as long as they attest)